### PR TITLE
Refactor all dialog inapp

### PR DESCRIPTION
**Pull request by nqhhdev**
### Issue: 

- In twake app, we use some routers: `FirstColumnInnerRoutes` and `GoRouter`
- We can't manage for `context` of both routers when displaying loading dialog on web, mobile, ...

### Resolved:

- Don't care to which router `context` belongs to because we will probably use multiple routers to build layers UI.
- Just only manage `Tweak Dialog` to display dialog using `context` of `GoRouter` because `context` of `Gouter` is the main router in twake app

https://github.com/linagora/twake-on-matrix/assets/99852347/70e8cca5-4491-4e6e-a830-c3314f19c795

### CODE REVIEW

**Review by drminh2807 — APPROVED** [review]

File: lib/utils/dialog/twake_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      useRootNavigator: false,
+    );
+  }
+
+  static Future<bool?> showCupertinoDialogFullScreen({
+    required Widget Function() builder,
+  }) {
+    final twakeContext = TwakeApp.routerKey.currentContext;
+    if (twakeContext == null) {
+      Logs().e(
```
Comment (sherlockvn) [inline lib/utils/dialog/twake_dialog.dart:79]: the `showCupertinoDialog` still called if twakeContext is null

File: lib/utils/dialog/twake_dialog.dart [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    return const AlertDialog(
+      content: Row(
+        children: [
+          Padding(
+            padding: EdgeInsets.only(right: 16.0),
+            child: CircularProgressIndicator.adaptive(),
+          ),
+          Expanded(
+            child: Text(
+              'Loading... Please Wait!',
```
Comment (sherlockvn) [inline lib/utils/dialog/twake_dialog.dart:106]: I think you should use l10n

File: lib/utils/dialog/twake_dialog.dart [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      future: future,
+    );
+  }
+
+  static Future<bool?> showDialogFullScreen({
+    required Widget Function() builder,
+  }) {
+    final twakeContext = TwakeApp.routerKey.currentContext;
+    if (twakeContext == null) {
+      Logs().e(
```
Comment (sherlockvn) [inline lib/utils/dialog/twake_dialog.dart:62]: same

File: lib/utils/dialog/twake_dialog.dart (hunk 1 above)
Comment (nqhhdev) [reply lib/utils/dialog/twake_dialog.dart:80]: done

File: lib/utils/dialog/twake_dialog.dart (hunk 2 above)
Comment (nqhhdev) [reply lib/utils/dialog/twake_dialog.dart]: done

File: lib/utils/dialog/twake_dialog.dart (hunk 3 above)
Comment (nqhhdev) [reply lib/utils/dialog/twake_dialog.dart:62]: done

**Review by hoangdat — APPROVED** [review]

**Review by sherlockvn — APPROVED** [review]

**Pull request merged**